src/library/scala/collection/immutable/Range.scala

```diff
@@ -165,6 +162,49 @@ sealed abstract class Range(
     }
   }
 
+  override final def indexOf[@sp(Int) B >: Int](elem: B, from: Int = 0): Int =
```

Comment:
porting this to `NumericRange` is nontrivial, because `B` is unbounded above; here we know the element type is `Int` (and can special-case it accordingly) but `NumericRange` has no way of telling whether `elem` is a `T` or not, so the `Integral` arithmetic is unsafe.